lib/tools/node-modules/node-modules-builder.ts

```diff
@@ -126,6 +126,14 @@ export class NodeModulesBuilder implements INodeModulesBuilder {
 		let dependenciesBuilder = this.$injector.resolve(NodeModulesDependenciesBuilder, {});
 		let productionDependencies = dependenciesBuilder.getProductionDependencies(this.$projectData.projectDir);
 
+		let prodDependenciesArr = [];
+		for (let i in productionDependencies) {
+			prodDependenciesArr.push(productionDependencies[i].name);
+		}
+
+		let prodDependenciesFilePath = path.join(this.$projectData.projectDir, "platforms", "android", "productionDependencies.json");
+		this.$fs.writeJson(prodDependenciesFilePath, prodDependenciesArr);
```

Comment:
Because gradle needs to go through the `node_modules` folder and get all `.aar` so it can include them into the build. Currently we traverse all of the `node_modules` which is error prone and it's a waste of time, when we know which dependencies we should use.